tough/src/editor/mod.rs

```diff
@@ -287,42 +389,217 @@ impl RepositoryEditor {
         self
     }
 
-    // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+    /// Takes the current Targets from `targets_editor` and inserts the role to its proper place in `signed_targets`
+    /// Sets `targets_editor` to None
+    /// Must be called before `change_delegated_targets()`
```

Comment:
Is there any security we can provide in `change_delegated_targets()`?  Relying on docstrings to deter this behavior seems a little scary!

Reply:
This doc string is warning that calling `change_delegated_targets()` without calling `signed_targets_editor()` will throw an error.